apps/minifront/src/state/ibc.ts

```diff
@@ -90,29 +90,57 @@ export const createIbcSendSlice = (): SliceCreator<IbcSendSlice> => (set, get) =
   };
 };
 
+// Timeout is two days. However, in order to prevent identifying oneself by clock skew,
+// timeout time is rounded to the nearest 10 minute mark.
+// Reference in core: https://github.com/penumbra-zone/penumbra/blob/1376d4b4f47f44bcc82e8bbdf18262942edf461e/crates/bin/pcli/src/command/tx.rs#L1066-L1067
+export const currentTimePlusTwoDaysRounded = (): bigint => {
+  // Get current time in milliseconds and convert to nanoseconds
+  const nowInNanoseconds = BigInt(Date.now()) * BigInt(1_000_000);
+
+  // New timestamp after adding 2 days
+  const twoDays = BigInt(2 * 24 * 60 * 60 * 1000 * 1000000);
+  const nowPlusTwoDays = nowInNanoseconds + twoDays;
+  const tenMinutes = BigInt(10 * 60 * 1000 * 1000000);
+
+  // Round up to nearest 10 minutes
+  return nowPlusTwoDays + tenMinutes - (nowPlusTwoDays % tenMinutes);
+};
```

Comment:
this is clearer for me to read.

* constants at the top, calculated with decreasing significance so it lines up
* name every variable with Ms or Ns depending on what it contains
* 'round up' confused me, i think 'next ten-minute interval' is clearer
* only *one* millis-to-nanos conversion, at the end
* use `_` separator on large integer

```suggestion
export const currentTimePlusTwoDaysRounded = (): bigint => {
  const tenMinsMs = BigInt(1000 * 60 * 10);
  const twoDaysMs = BigInt(1000 * 60 * 60 * 24 * 2);
  
  // Date.now returns milliseconds
  const nowMs = BigInt(Date.now());
  const twoDaysFromNowMs = nowMs + twoDaysMs;
  
  // round to next ten-minute interval
  const roundedTimeoutMs = twoDaysFromNowMs + tenMinsMs - (twoDaysFromNowMs % tenMinsMs);
  
  // 1 million nanoseconds per millisecond
  const roundedTimeoutNs = roundedTimeoutMs * BigInt(1_000_000);
  return roundedTimeoutNs;
};
```

Reply:
i would also suggest accepting an input parameter, and having the caller provide Date.now()

Reply:
Updated to your format. However, changed so that bigint conversion happens at the end.